pkg/controller/validate_config.go

```diff
@@ -104,3 +111,78 @@ func (c *ValidateConfigHandler) Handle(ctx context.Context) {
 	ctx = CtxClusterStatus.WithValue(ctx, currentStatus)
 	c.next.Handle(ctx)
 }
+
+func (c *ValidateConfigHandler) getAvailableVersions(ctx context.Context, graph updates.UpdateGraph, version v1alpha1.SpiceDBVersion, datastore string) []v1alpha1.SpiceDBVersion {
+	logger := logr.FromContextOrDiscard(ctx)
+	source, err := graph.SourceForChannel(version.Channel)
+	if err != nil {
+		logger.V(4).Error(err, "no source found for channel %q, can't compute available versions", version.Channel)
+	}
+
+	availableVersions := make([]v1alpha1.SpiceDBVersion, 0)
+	nextDirect := source.NextDirect(version.Name)
```

Comment:
given the risks of passing the wrong items into these functions, I wonder if we should make them take in a `Version(...)` type or `Channel(...)` type, to make it explicit at the call site what type of string is being passed in